discovery_engine_core/core/src/ranker.rs

```diff
@@ -36,6 +36,9 @@ pub trait Ranker {
     /// Performs the ranking of [`Document`] items.
     fn rank(&mut self, items: &mut [Document]) -> Result<(), GenericError>;
 
+    /// Performs the ranking of [`TrendTopic`] items.
+    fn rank_topics(&mut self, items: &mut [TrendTopic]) -> Result<(), GenericError>;
```

Comment:
🎨 We could also modify `rank` above to accept all `ranker::Document`s:

```
fn rank<T>(&mut self, items: &mut [T]) -> Result<(), GenericError>
    where
      T: xayn_ai::ranker::Document,
```

Reply:
The disadvantage would be that this trait would no longer be object safe. But there is also the question of whether it has to be object safe.

Reply:
Fair point. I don't think it would currently be relevant, and I wouldn't be able to gauge whether it could be in the future.

Reply:
it doesn't seem to much disadvantage for now (there was one place that required a type annotation, but that was it) so I will take this suggestion in, thanks.